### DISCUSSION TITLE
Macos系统使用chatglm-6b-int4模型，☹️发生了错误：“LayerNormKernelImpl” not implemented for ‘Half’，如何解决。

### DISCUSSION THREAD

**Discussion by sunchch1020** [GaiZhenbiao/ChuanhuChatGPT#649] [Q&A]
Macos系统使用chatglm-6b-int4模型，☹️发生了错误：“LayerNormKernelImpl” not implemented for ‘Half’，如何解决。

**Comment by GaiZhenbiao** [comment]
请升级到最新版代码，应该解决了这个问题

**Reply by sunchch1020** [reply]
麻烦再问下，本地导入了chatglm模型，也需要在网络环境下才能运行吗？断网的情况下我看是无法运行的。还有，macos系统无法使用cuda，有其他方法吗？感谢回答

**Reply by GaiZhenbiao** [reply]
可以断网运行的。不过，index功能仍需联网。也可以使用在config里设置local_embedding为true，这样index也可以本地运行了。

**Reply by LeeKonyu** [reply]
> 麻烦再问下，本地导入了chatglm模型，也需要在网络环境下才能运行吗？断网的情况下我看是无法运行的。还有，macos系统无法使用cuda，有其他方法吗？感谢回答

麻烦问一下，我刚开始接触这方面的知识，我本地已经部署了chatglm，也下好了chuanhu，我要怎么才能导入chatglm模型呀。非常感谢，孩子太急了😭
